jcl/src/openj9.sharedclasses/share/classes/com/ibm/oti/shared/SharedClassAbstractHelper.java

```diff
@@ -94,17 +94,35 @@ private String recursiveJarTrim(String jarName) {
 		if (jarName==null)
 			return null;
 
-		int lastIndexBang = jarName.lastIndexOf("!/"); //$NON-NLS-1$
 		boolean startsWithJar = jarName.startsWith("jar:"); //$NON-NLS-1$
+		boolean endsWithBang = jarName.endsWith("!/");//$NON-NLS-1$
```

Comment:
Please add comments to explain the reasoning for using endsWith().

Reply:
A comment is added two lines after this

Reply:
The comment below doesn't explain why `!/` is only stripped when its at the end, and not in the middle. The code makes sense in the context of this change, adding another comment is to help future readers understand when looking at the code as a whole outside of the context of this change.

Reply:
More comment added